build.sbt

```diff
@@ -49,6 +49,10 @@ lazy val core = crossProject(JVMPlatform, JSPlatform, NativePlatform)
         Seq("-Xasync")
     },
 
+    tlFatalWarnings := {
+      tlFatalWarnings.value && !tlIsScala3.value
+    },
```

Comment:
idk if it's legit or not, would be ideal to fix this and re-enable fatals.
```
[warn] -- Warning: /workspace/cats-effect-cps/core/shared/src/test/scala/cats/effect/cps/AsyncAwaitSpec.scala:217:25 
[warn] 217 |        async[OptionTIO](optionT.await).value.await
[warn]     |                         ^^^^^^^^^^^^^
[warn]     |                         unused variable creation, effect may be lost
[warn]     |---------------------------------------------------------------------------
[warn]     |Inline stack trace
[warn]     |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
[warn]     |This location contains code that was inlined from AsyncAwaitSpec.scala:217
[warn]      ---------------------------------------------------------------------------
[warn] one warning found
```

Reply:
As far as I can tell that one is out of our direct control, most likely happening during the macro expansion carried out by `dotty-cps-async`. 

Maybe @rssh has an idea ? Ruslan, if you lurk, the repro steps are, once this branch is checked out and SBT is started : 

``` 
> ++ 3.2.1 
> coreJVM/test 
```

That being said, I don't think this is an impediment to the merging of this PR

Reply:
You can define
```
implicit val printCode = cps.macros.flags.PrintCode
```
to see,  what's going on

Reply:
The problem that in internal async we materialize monad in var, which is not used (and issue warning about this) because the result is await which translated to argument.